[2024-06-16T00:18:33.091+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-16T00:18:33.169+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.generate_summary manual__2024-06-16T00:15:29+00:00 [queued]>
[2024-06-16T00:18:33.192+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Air_pollution_data_pipeline.generate_summary manual__2024-06-16T00:15:29+00:00 [queued]>
[2024-06-16T00:18:33.193+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 6
[2024-06-16T00:18:33.226+0000] {taskinstance.py:2330} INFO - Executing <Task(_PythonDecoratedOperator): generate_summary> on 2024-06-16 00:15:29+00:00
[2024-06-16T00:18:33.242+0000] {standard_task_runner.py:63} INFO - Started process 3912 to run task
[2024-06-16T00:18:33.253+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Air_pollution_data_pipeline', 'generate_summary', 'manual__2024-06-16T00:15:29+00:00', '--job-id', '107', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/tmp/tmpnc45nitv']
[2024-06-16T00:18:33.261+0000] {standard_task_runner.py:91} INFO - Job 107: Subtask generate_summary
[2024-06-16T00:18:33.590+0000] {task_command.py:426} INFO - Running <TaskInstance: Air_pollution_data_pipeline.generate_summary manual__2024-06-16T00:15:29+00:00 [running]> on host 6e26965f02f1
[2024-06-16T00:18:37.261+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='khalid' AIRFLOW_CTX_DAG_ID='Air_pollution_data_pipeline' AIRFLOW_CTX_TASK_ID='generate_summary' AIRFLOW_CTX_EXECUTION_DATE='2024-06-16T00:15:29+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-16T00:15:29+00:00'
[2024-06-16T00:18:37.265+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-16T00:18:37.267+0000] {summary_analysis.py:13} INFO - starting summary statistics of pollutants emission
[2024-06-16T00:18:37.344+0000] {logging_mixin.py:188} WARNING - /opt/***/dags/statistical_analysis/summary_analysis.py:16 FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
[2024-06-16T00:18:45.517+0000] {summary_analysis.py:22} INFO - successfully calculated pollutants emission summary statistics
[2024-06-16T00:18:45.518+0000] {summary_analysis.py:26} ERROR - error calculating pollutants summary statistics
Traceback (most recent call last):
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 23, in generate_summary
    return sum_stat.tojson()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/generic.py", line 6299, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'tojson'
[2024-06-16T00:18:45.526+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-16T00:18:45.528+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/decorators/base.py", line 265, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/statistical_analysis/summary_analysis.py", line 23, in generate_summary
    return sum_stat.tojson()
  File "/home/airflow/.local/lib/python3.10/site-packages/pandas/core/generic.py", line 6299, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'DataFrame' object has no attribute 'tojson'
[2024-06-16T00:18:45.549+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=Air_pollution_data_pipeline, task_id=generate_summary, run_id=manual__2024-06-16T00:15:29+00:00, execution_date=20240616T001529, start_date=20240616T001833, end_date=20240616T001845
[2024-06-16T00:18:45.580+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 107 for task generate_summary ('DataFrame' object has no attribute 'tojson'; 3912)
[2024-06-16T00:18:45.615+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-16T00:18:45.690+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-16T00:18:45.706+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
